packages/trpc/src/router/site/validations.ts

```diff
@@ -0,0 +1,73 @@
+import { z } from 'zod';
+
+export const SITE_TEMPLATE_URL_VALIDATION = z
+  .string({
+    required_error: 'Notion template URL is required',
+  })
+  .superRefine((val, ctx) => {
+    let url: URL;
+    try {
+      url = new URL(val);
+    } catch {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'Invalid URL',
+        fatal: true,
+      });
+      return z.NEVER;
+    }
+    if (url.hostname.endsWith(`.notion.so`)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: `The Notion page isn't public, you need to share it to web first`,
+        fatal: true,
+      });
+      return z.NEVER;
+    }
+    if (!url.hostname.endsWith(`.notion.site`)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'Invalid Notion URL, domain should be notion.site',
+        fatal: true,
+      });
+      return z.NEVER;
+    }
+    const pageId = url.pathname.split('-').pop();
+    if (!pageId || !/^[\da-z]+$/.test(pageId)) {
+      ctx.addIssue({
+        code: z.ZodIssueCode.custom,
+        message: 'Invalid Notion URL, page ID is missing or invalid',
+        fatal: true,
+      });
+      return z.NEVER;
+    }
+  });
+
+export const SITE_NAME_VALIDATION = z
+  .string()
+  .min(1, 'Name must contain at least 1 character')
+  .max(190, 'Name must contain at most 190 characters');
```

Comment:
Shall we also show `x/190` in the UX to tell user the limit?

Reply:
Usually, a user can't reach this limit, we only show it when it happens. UI hint seems redundant to me